packages/plans-grid-next/src/components/actions.tsx

```diff
@@ -290,30 +256,22 @@ const PlanFeatures2023GridActions = ( {
 	return (
 		<div className="plan-features-2023-gridrison__actions">
 			<div className="plan-features-2023-gridrison__actions-buttons">
-				{ isInSignup || isWpcomEnterpriseGridPlan( planSlug ) ? (
-					<PlanFeatureActionButton
-						planSlug={ planSlug }
-						isStuck={ isStuck }
-						postButtonText={ postButtonText }
-						status={ status }
-						hasFreeTrialPlan={ isInSignup ? !! freeTrialPlanSlug : false }
-						onCtaClick={ callback }
-						onFreeTrialCtaClick={ freeTrialCallback }
-						text={ text }
-						freeTrialText={ freeTrialText }
-					/>
-				) : (
-					<LoggedInPlansFeatureActionButton
-						disabled={ ! callback || 'disabled' === status }
-						planSlug={ planSlug }
-						availableForPurchase={ availableForPurchase }
-						onCtaClick={ callback }
-						currentSitePlanSlug={ currentSitePlanSlug }
-						isMonthlyPlan={ isMonthlyPlan }
-						storageOptions={ storageOptions }
-						text={ text }
-					/>
-				) }
+				<PlanFeatureActionButton
```

Comment:
I think let's flatten this out more -> so remove `PlanFeatureActionButton` and keep all the logic under `PlanFeatures2023GridActions`. It's a largely unnecessary (redundant) abstraction that ultimately just makes it harder to move more things into `useAction` hook. 

Keeping the ideal end state in perspective: there will be no (zero) conditions in this file.

Reply:
Great callout I'll make it happen 👍

Reply:
[cf20b24](https://github.com/Automattic/wp-calypso/pull/90611/commits/cf20b24aec22827035f53fd29fb7b4e0f5a9fe35)